Sources/Sentry/SentryPredicateDescriptor.m

```diff
@@ -0,0 +1,136 @@
+#import "SentryPredicateDescriptor.h"
+
+@implementation SentryPredicateDescriptor
+
+- (NSString *)predicateDescription:(NSPredicate *)predicate
+{
+    if ([predicate isKindOfClass:[NSCompoundPredicate class]]) {
+        return [self compoundPredicateDescription:(NSCompoundPredicate *)predicate];
+    } else if ([predicate isKindOfClass:[NSComparisonPredicate class]]) {
+        return [self comparisonPredicateDescription:(NSComparisonPredicate *)predicate];
+    } else if ([predicate isKindOfClass:[NSExpression class]]) {
+        return [self expressionDescription:(NSExpression *)predicate];
+    }
+
+    return @"<UNKNOWN>";
+}
+
+- (NSString *)compoundPredicateDescription:(NSCompoundPredicate *)predicate
+{
+
+    NSMutableArray<NSString *> *expressions =
+        [[NSMutableArray alloc] initWithCapacity:predicate.subpredicates.count];
+
+    for (NSPredicate *sub in predicate.subpredicates) {
+        if ([sub isKindOfClass:[NSCompoundPredicate class]]) {
+            [expressions
+                addObject:[NSString stringWithFormat:@"(%@)", [self predicateDescription:sub]]];
+        } else {
+            [expressions addObject:[self predicateDescription:sub]];
+        }
+    }
+
+    if (expressions.count == 1) {
+        return [NSString stringWithFormat:@"%@ %@",
+                         [self compoundPredicateTypeDescription:predicate.compoundPredicateType],
+                         expressions.firstObject];
+    }
+
+    return [expressions
+        componentsJoinedByString:[self
+                                     compoundPredicateTypeDescription:predicate
+                                                                          .compoundPredicateType]];
+}
+
+- (NSString *)comparisonPredicateDescription:(NSComparisonPredicate *)predicate
+{
+    NSString *operator= [self predicateOperatorTypeDescription:predicate.predicateOperatorType];
+
+    if (operator== nil) {
+        return @"<COMPARISON NOT SUPPORTED>";
+    }
+
+    return [NSString stringWithFormat:@"%@ %@ %@",[self expressionDescription:predicate.leftExpression] ,
+            operator,
+            [self expressionDescription:predicate.rightExpression]
+    ];
+}
+
+- (NSString *)expressionDescription:(NSExpression *)predicate
+{
+    switch (predicate.expressionType) {
+    case NSConstantValueExpressionType:
+        return @"%@";
+    case NSAggregateExpressionType:
+        if ([predicate.collection isKindOfClass:[NSArray class]]) {
```

Comment:
`m`: Could there also be an `NSDictionary`?

Reply:
I dont see how this could be a NSDictionary. I dont think core data supports dictionaries.